src/Microsoft.AspNetCore.Routing/Matchers/MatchProcessor.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    public abstract class MatchProcessor
```

Comment:
I'd like to revisit this name in the future.

Reply:
Yeah right. It would end with "...ConstraintProcesor" I think.